akka-jackson/src/main/scala/akka/jackson/JacksonSerializer.scala

```diff
@@ -0,0 +1,200 @@
+import java.io.NotSerializableException
+import java.util.zip.GZIPInputStream
+import java.util.zip.GZIPOutputStream
+
+import scala.annotation.tailrec
+import scala.util.Failure
+import scala.util.Success
+
+import akka.actor.ExtendedActorSystem
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.serialization.BaseSerializer
+import akka.serialization.SerializerWithStringManifest
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.cbor.CBORFactory
+import com.fasterxml.jackson.dataformat.smile.SmileFactory
+
+/**
+ * INTERNAL API: only public by configuration
+ *
+ * Akka serializer for Jackson with JSON.
+ */
+@InternalApi private[akka] class JacksonJsonSerializer(system: ExtendedActorSystem)
+  extends JacksonSerializer(system, JacksonObjectMapperProvider.create(system, None))
+
+/**
+ * INTERNAL API: only public by configuration
+ *
+ * Akka serializer for Jackson with Smile.
+ */
+@InternalApi private[akka] class JacksonSmileSerializer(system: ExtendedActorSystem)
+  extends JacksonSerializer(system, JacksonObjectMapperProvider.create(system, Some(new SmileFactory)))
+
+/**
+ * INTERNAL API: only public by configuration
+ *
+ * Akka serializer for Jackson with CBOR.
+ */
+@InternalApi private[akka] class JacksonCborSerializer(system: ExtendedActorSystem)
+  extends JacksonSerializer(system, JacksonObjectMapperProvider.create(system, Some(new CBORFactory)))
+
+/**
+ * INTERNAL API: Base class for Jackson serializers.
+ *
+ * Configuration in `akka.jackson` section.
+ * It will load Jackson modules defined in configuration `jackson-modules`.
+ *
+ * It will compress the payload if the the payload is larger than the configured
+ * `compress-larger-than` value.
+ */
+@InternalApi private[akka] abstract class JacksonSerializer(
+  val system: ExtendedActorSystem, objectMapper: ObjectMapper)
+  extends SerializerWithStringManifest with BaseSerializer {
+
+  private val log = Logging.getLogger(system, getClass)
+  private val conf = system.settings.config.getConfig("akka.jackson")
+  private val isDebugEnabled = conf.getBoolean("verbose-debug-logging") && log.isDebugEnabled
+  private final val BufferSize = 1024 * 4
+  private val migrations: Map[String, JacksonMigration] = {
+    import scala.collection.JavaConverters._
+    conf.getConfig("migrations").root.unwrapped.asScala.toMap.map {
+      case (k, v) ⇒
+        val transformer = system.dynamicAccess.createInstanceFor[JacksonMigration](v.toString, Nil).get
+        k -> transformer
+    }(collection.breakOut)
+  }
+
+  private val compressLargerThan: Long = conf.getBytes("compress-larger-than")
+
+  override def manifest(obj: AnyRef): String = {
+    val className = obj.getClass.getName
+    migrations.get(className) match {
+      case Some(transformer) ⇒ className + "#" + transformer.currentVersion
+      case None              ⇒ className
+    }
+  }
+
+  override def toBinary(obj: AnyRef): Array[Byte] = {
```

Comment:
Would be interesting to see if we can use/if it provides any additional gain to use the ByteBufferSerializer api introduced when we wrote artery  (I used that when playing around with jackson last week, jackson has some helper input/output stream utils to write to buffers)